libs/cluster/Server/ClusterProvider.cs

```diff
@@ -119,6 +119,19 @@ public void Dispose()
         public bool IsReplica()
             => clusterManager?.CurrentConfig.LocalNodeRole == NodeRole.REPLICA || replicationManager?.Recovering == true;
 
+        /// <inheritdoc />
+        public bool IsReplica(string nodeId)
+        {
+            var config = clusterManager?.CurrentConfig;
+            if (config is null || !config.IsKnown(nodeId))
+            {
+                return false;
+            }
+
+            var worker = config.GetWorkerFromNodeId(nodeId);
+            return worker.Role == NodeRole.REPLICA;
+        }
+
```

Comment:
It might be better to use GetNodeRoleFromNodeId(nodeId) to avoid having to do two scans of the Workers array.

```suggestion
        public bool IsReplica(string nodeId)
        {
            var config = clusterManager?.CurrentConfig;
            return config is null || config.GetNodeRoleFromNodeId(nodeId) != NodeRole.REPLICA
        }
```

Reply:
Ah, I missed that `workers` is padded with a dead value so the `IsKnown(...)` check is unnecessary.